Library/Homebrew/formula.rb

```diff
@@ -405,11 +413,19 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
+  def latest_head_prefix
+    head_prefixes = installed_prefixes.select do |pn|
+      pn.basename.to_s.start_with?("HEAD")
```

Comment:
I think for this PR, we just assume head prefix is always up to dated(just like homebrew has always assumed before this PR). Therefore, you don't need to consider `head_prefix_up_to_date` at all in this PR. As result, brew install/reinstall/cleanup should work fine. The only thing missing is brew upgrade, which should be targeted in the other PR.